### PR TITLE
[Wishlist] Remove name from WishlistOutput #920

### PR DESCRIPTION
**Pull request by XxXgeoXxX**
### Description (*)
Original issue: https://github.com/magento/graphql-ce/issues/920

### Questions or comments
1. Should we create separate resolver for Customer instead use Wishlist resolver?
2. Should we deprecated query 'Wishlists'?

Note: I'm working to coverage this mutation by functional tests.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/WishlistGraphQl/Model/Resolver/CustomerWishlistsResolver.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        $customerId = $context->getUserId();
+
+        /* Guest checking */
+        if (!$customerId && 0 === $customerId) {
+            throw new GraphQlAuthorizationException(__('The current user cannot perform operations on wishlist'));
+        }
+        $collection = $this->_wishlistCollectionFactory->create()->filterByCustomerId($customerId);
+        $wishlists = $collection->getItems();
+        $wishlistsData = [];
+        if (0 === count($wishlists)) {
```
Comment (lenaorobei) [inline app/code/Magento/WishlistGraphQl/Model/Resolver/CustomerWishlistsResolver.php]: Please use `getSize` method instead.

File: app/code/Magento/WishlistGraphQl/Model/Resolver/CustomerWishlistsResolver.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $customerId = $context->getUserId();
+
+        /* Guest checking */
+        if (!$customerId && 0 === $customerId) {
```
Comment (lenaorobei) [inline app/code/Magento/WishlistGraphQl/Model/Resolver/CustomerWishlistsResolver.php]: `0 === $customerId` is already covered by `!$customerId` check. Please make check more specific.

File: app/code/Magento/WishlistGraphQl/etc/schema.graphqls [hunk 3]
```diff
@@ -5,10 +5,21 @@ type Query {
     wishlist: WishlistOutput @resolver(class: "\\Magento\\WishlistGraphQl\\Model\\Resolver\\WishlistResolver") @doc(description: "The wishlist query returns the contents of a customer's wish list") @cache(cacheable: false)
 }
 
-type WishlistOutput {
+type Customer {
+    wishlists: [Wishlist]! @resolver(class:"\\Magento\\WishlistGraphQl\\Model\\Resolver\\CustomerWishlistsResolver") @doc(description: "The wishlist query returns the contents of a customer's wish lists")  @cache(cacheable: false)
+}
+
+type WishlistOutput @doc(description: "Deprecated: 'Wishlist' type should be used instead") {
+     items: [WishlistItem] @deprecated(reason: "Use field `items` from type `Wishlist` instead") @resolver(class: "\\Magento\\WishlistGraphQl\\Model\\Resolver\\WishlistItemsResolver") @doc(description: "An array of items in the customer's wish list"),
```
Comment (lenaorobei) [inline app/code/Magento/WishlistGraphQl/etc/schema.graphqls]:
```suggestion
    items: [WishlistItem] @deprecated(reason: "Use field `items` from type `Wishlist` instead") @resolver(class: "\\Magento\\WishlistGraphQl\\Model\\Resolver\\WishlistItemsResolver") @doc(description: "An array of items in the customer's wish list"),
```

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Please move the logic for getting customer wish list to the separate class with `execute` method and use it in both places - where array of wishlists and single wish list is returned.

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-6165](https://jira.corp.magento.com/browse/ENGCOM-6165) has been created to process this Pull Request
:eight_spoked_asterisk: @lenaorobei, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by lenaorobei** [comment]
fixed https://github.com/magento/graphql-ce/issues/1038

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-6165](https://jira.corp.magento.com/browse/ENGCOM-6165) has been created to process this Pull Request

**Pull request merged**